lib/Doctrine/DBAL/Platforms/SQLServer2012Platform.php

```diff
@@ -120,9 +120,18 @@ protected function doModifyLimitQuery($query, $limit, $offset = null)
         if ($orderByPos === false
             || substr_count($query, "(", $orderByPos) - substr_count($query, ")", $orderByPos)
         ) {
-            // In another DBMS, we could do ORDER BY 0, but SQL Server gets angry if you use constant expressions in
-            // the order by list.
-            $query .= " ORDER BY (SELECT 0)";
+            if (stripos($query, 'SELECT DISTINCT') === 0) {
+                // SQL Server won't let us order by a non-selected column in a DISTINCT query,
+                // so we have to do this madness. This says, order by the first column in the
```

Comment:
There is, but there is no existing functional test suite for DBAL that covers this kind of stuff, and I don't have time to write one. Ideally we should have a testsuite that covers actually executing these queries on all platforms, but we don't :(

I've been using the ORM as my integration testsuite for DBAL, bad as that is.